### PR TITLE
ci: packaging pipeline for the snapshots

### PR DESCRIPTION
**Pull request by v1v**
## What is the problem this PR solves?

Enable the Daily Release Artifacts for the Fleet Server project (Snapshots only)

If something bad happens then it sends an email/slack to:
- `#fleet-server`
- `fleet-server+package@...`

## How does this PR solve the problem?

A specific pipeline in charge of publishing the snapshots on a merge-commit only

## Related issues

Requires https://github.com/elastic/fleet-server/pull/1233
Requires https://github.com/elastic/apm-pipeline-library/pull/1618

## Test

https://artifacts-snapshot.elastic.co/fleet-server/8.2.0-6e3e8531/summary-8.2.0-SNAPSHOT.html has been generated (see [build](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Ffleet-server-package-mbp/detail/PR-1234/46/pipeline/262))

image

## Follow ups

Support BCs (to be done later on)

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        stage('Package') {
+          options { skipDefaultCheckout() }
+          matrix {
+            agent {
+              label "${PLATFORM}"
+            }
+            axes {
+              axis {
+                name 'PLATFORM'
+                values 'ubuntu-20 && immutable', 'arm'
```
Comment (v1v) [inline .ci/packaging.groovy:81]: As long as `zip` is not available in the cache images, let's try this

```suggestion
                values 'ubuntu-20 && immutable && gobld/image:family/elastic-beats-ci-ubuntu-2004-lts-nocache', 'arm'
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+    failure {
+      echo 'disabled'
+    // notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}] DRA failed", body: "Build: (<${env.RUN_DISPLAY_URL}|here>)")
```
Comment (v1v) [inline .ci/packaging.groovy:153]: TBR

File: .ci/scripts/release-manager.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+docker run --rm \
+  --name release-manager \
+  -e VAULT_ADDR \
+  -e VAULT_ROLE_ID \
+  -e VAULT_SECRET_ID \
+  --mount type=bind,readonly=false,src="$PWD",target=/artifacts \
+  "$IMAGE" \
+    cli collect \
+      --project fleet-server \
+      --branch "main" \
```
Comment (v1v) [inline .ci/scripts/release-manager.sh:30]: TBR

File: .ci/packaging.groovy [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        stage('Package') {
+          options { skipDefaultCheckout() }
+          matrix {
+            agent {
+              label "${PLATFORM}"
+            }
+            axes {
+              axis {
+                name 'PLATFORM'
+                values 'ubuntu-20 && immutable && gobld/image:family/elastic-beats-ci-ubuntu-2004-lts-nocache', 'arm'
```
Comment (v1v) [inline .ci/packaging.groovy:83]:
```suggestion
                values 'ubuntu-20 && immutable', 'arm'
```

**Comment by ph** [comment]
@v1v lets us know when you want a review on this.

File: .ci/packaging.groovy [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult(prComment: false)
+    }
+    failure {
+      echo 'disabled'
+    // notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}] DRA failed", body: "Build: (<${env.RUN_DISPLAY_URL}|here>)")
```
Comment (v1v) [inline .ci/packaging.groovy:149]:
```suggestion
      notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}] DRA failed", body: "Build: (<${env.RUN_DISPLAY_URL}|here>)")
```

File: .ci/scripts/release-manager.sh (hunk 3 above)
Comment (v1v) [reply .ci/scripts/release-manager.sh:33]:
```suggestion
      --branch "$BRANCH_NAME" \
```

File: .ci/scripts/release-manager.sh [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+# - VAULT_ROLE_ID
+# - VAULT_SECRET_ID
+#
+set -uexo pipefail
+
+source /usr/local/bin/bash_standard_lib.sh
+
+# set required permissions on artifacts and directory
+chmod -R a+r build/*
+chmod -R a+w build
```
Comment (v1v) [inline .ci/scripts/release-manager.sh:16]: We won't need this if the report dependencies to be consumed by the Unified Release is stored in the build/distributions folder

```suggestion
chmod -R a+r build/distributions/*
chmod -R a+w build/distributions
```

File: .ci/packaging.groovy [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                  PACKAGES = "${isArm() ? 'docker' : ''}"
+                  SNAPSHOT = 'true'
+                }
+                steps {
+                  deleteDir()
+                  unstash 'source'
+                  dir("${BASE_DIR}"){
+                    withMageEnv() {
+                      whenFalse(isArm()) {
+                        sh(label: 'make build/reports/dependencies.csv', script: 'make build/reports/dependencies.csv')
```
Comment (v1v) [inline .ci/packaging.groovy:99]:
```suggestion
                        sh(label: 'make build/distributions/dependencies.csv', script: 'make build/distributions/dependencies.csv')
```

File: .ci/packaging.groovy [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult(prComment: false)
+    }
+    failure {
+      echo 'disabled'
+    // notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}@${env.BRANCH_NAME}] DRA failed", body: "Build: (<${env.RUN_DISPLAY_URL}|here>)")
```
Comment (v1v) [inline .ci/packaging.groovy:153]:
```suggestion
      notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}@${env.BRANCH_NAME}] DRA failed", body: "Build: (<${env.RUN_DISPLAY_URL}|here>)")
```

File: .ci/packaging.groovy [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                        shallow: false, reference: "/var/lib/jenkins/.git-references/${REPO}.git")
+            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+            // set environment variables globally since they are used afterwards but GIT_BASE_COMMIT won't
+            // be available until gitCheckout is executed.
+            setEnvVar('URI_SUFFIX', "commits/${env.GIT_BASE_COMMIT}")
+            // JOB_GCS_BUCKET contains the bucket and some folders, let's build the folder structure
+            setEnvVar('PATH_PREFIX', "${JOB_GCS_BUCKET.contains('/') ? JOB_GCS_BUCKET.substring(JOB_GCS_BUCKET.indexOf('/') + 1) + '/' + env.URI_SUFFIX : env.URI_SUFFIX}")
+            //TODO : uncomment
+            //setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
+            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable('main'))
```
Comment (v1v) [inline .ci/packaging.groovy:73]:
```suggestion
            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
```

**Review by cachedout — APPROVED** [review]

File: .ci/packaging.groovy [hunk 10]
```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env groovy
+@Library('apm@current') _
+
+pipeline {
+  agent none
+  environment {
+    REPO = 'fleet-server'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    SLACK_CHANNEL = '#fleet-server'
+    NOTIFY_TO = 'package+fleet-server@elastic.co'
```
Comment (v1v) [inline .ci/packaging.groovy:10]:
```suggestion
    NOTIFY_TO = 'fleet-server+build-package@elastic.co'
```

**Comment by v1v** [comment]
@mergifyio backport 8.1

**Pull request merged**